### PR TITLE
feat: add filter and sort on slonik

### PR DESCRIPTION
**Pull request by surajrai-dzango**

### CODE REVIEW

**Review by rameshlohala — APPROVED** [review]

File: packages/slonik/src/sql.ts [hunk 1]
```diff
@@ -18,4 +21,39 @@ const createWhereIdFragment = (id: number | string) => {
   return sql`WHERE id = ${id}`;
 };
 
-export { createLimitFragment, createTableFragment, createWhereIdFragment };
+const createFilterFragment = (
```
Comment (opichon) [inline packages/slonik/src/sql.ts:24]: This method is not in alphabetical order

File: packages/slonik/src/sql.ts [hunk 2]
```diff
@@ -18,4 +21,39 @@ const createWhereIdFragment = (id: number | string) => {
...
+  }
+
+  return sql`ORDER BY id ASC`;
+};
+
+export {
+  createLimitFragment,
+  createTableFragment,
+  createWhereIdFragment,
+  createFilterFragment,
```
Comment (opichon) [inline packages/slonik/src/sql.ts:57]: Wrong alphabetical order

File: packages/slonik/src/sqlFactory.ts [hunk 3]
```diff
@@ -1,4 +1,10 @@
-import { createLimitFragment, createTableFragment } from "./sql";
+import {
+  createLimitFragment,
+  createTableFragment,
+  createFilterFragment,
```
Comment (opichon) [inline packages/slonik/src/sqlFactory.ts:4]: Wrong alphabetical order

File: packages/slonik/src/dbFilters.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
+import { QueryResultRow, TaggedTemplateLiteralInvocation, sql } from "slonik";
```
Comment (opichon) [inline packages/slonik/src/dbFilters.ts:1]: Why not call this `filters.ts`. The `db` prefix is not required, and in any case it's an abbreviation, which we anoid.

File: packages/slonik/src/dbFilters.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      clauseOperator = not ? sql`NOT BETWEEN` : sql`BETWEEN`;
+      value = sql`${sql.join(value.split(","), sql` AND `)}`;
+      break;
+    }
+  }
+
+  return sql`${databaseField} ${clauseOperator} ${value}`;
+};
+
+const applyFiltersToQuery = (
```
Comment (opichon) [inline packages/slonik/src/dbFilters.ts:65]: Inconsistent naming of methods:`applyFiltersToQuery` vs `applyFilter`. For reasons of consistency and simplicity, rename this method to `applyFilters`.

File: packages/slonik/src/dbFilters.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+  filters: FilterInput,
+  tableName: string,
+  not = false
+) => {
+  const andFilter: TaggedTemplateLiteralInvocation<QueryResultRow>[] = [];
+  const orFilter: TaggedTemplateLiteralInvocation<QueryResultRow>[] = [];
+  let queryFilter;
```
Comment (opichon) [inline packages/slonik/src/dbFilters.ts:72]: add blank line

File: packages/slonik/src/dbFilters.ts [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+  const applyFilters = (
```
Comment (opichon) [inline packages/slonik/src/dbFilters.ts:74]: So the `applyFiltersToQuery method creates and invokes a `applyFilters` method? Why not just run the code?

File: packages/slonik/src/dbFilters.ts (hunk 5 above)
Comment (opichon) [reply packages/slonik/src/dbFilters.ts:65]: Also "apply" is not quite appropriate here. The method creates the SQL that will be used to apply the filters when the query is run.

**Pull request merged**